### PR TITLE
[msys2 add] disable-download-timeout & overwrite

### PR DESCRIPTION
**Pull request by MVoz**
+add full update dist

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@strega-nil, could you help futhre review?

File: scripts/cmake/vcpkg_acquire_msys.cmake [hunk 1]
```diff
@@ -92,7 +92,7 @@ function(vcpkg_acquire_msys PATH_TO_ROOT_OUT)
       WORKING_DIRECTORY ${TOOLPATH}
     )
     _execute_process(
-      COMMAND ${PATH_TO_ROOT}/usr/bin/bash.exe --noprofile --norc -c "PATH=/usr/bin;pacman -Syu --noconfirm"
+      COMMAND ${PATH_TO_ROOT}/usr/bin/bash.exe --noprofile --norc -c "PATH=/usr/bin;pacman -Syuu --noconfirm --disable-download-timeout --overwrite '*'"
```
Comment (strega-nil) [inline scripts/cmake/vcpkg_acquire_msys.cmake:95]: What's `-Syuu`? I couldn't find mention on pacman's manpage.

**Comment by Voskrese** [comment]
full update

File: scripts/cmake/vcpkg_acquire_msys.cmake (hunk 1 above)
Comment (Voskrese) [reply scripts/cmake/vcpkg_acquire_msys.cmake:95]: `-Syuu` - full update

although I don't know how the "CI" will behave, a BASH reload may be required

Comment (strega-nil) [reply scripts/cmake/vcpkg_acquire_msys.cmake:95]: Could you add a timeout to `_execute_process`, and just set it to like... I dunno, an hour or something?

Comment (Voskrese) [reply scripts/cmake/vcpkg_acquire_msys.cmake:95]: It's not required, with a full update can update itself BASH, pacman

When the bash is running, you get a message that you need to restart it to successfully upgrade

or you're about another?

Comment (strega-nil) [reply scripts/cmake/vcpkg_acquire_msys.cmake:95]: I mean, we don't want to just not timeout if there's an infinite loop somewhere.

Comment (Voskrese) [reply scripts/cmake/vcpkg_acquire_msys.cmake:95]: unfortunately i cannot edit
add

    _execute_process(
      COMMAND ${PATH_TO_ROOT}/usr/bin/bash.exe --noprofile --norc -c "PATH=/usr/bin;pacman -Syuu --noconfirm --disable-download-timeout --overwrite '*'"
      WORKING_DIRECTORY ${TOOLPATH}
    )
```
    _execute_process(
      COMMAND ${CMAKE_COMMAND} -E sleep 3600
    )
```
    file(WRITE "${TOOLPATH}/${STAMP}" "0")

**or**

```
function(sleep delay)
  execute_process(
    COMMAND ${CMAKE_COMMAND} -E sleep ${delay}
    RESULT_VARIABLE result
    )
  if(NOT result EQUAL 0)
    message(FATAL_ERROR "failed to sleep for ${delay} second.")
  endif()
endfunction(sleep)
```

`sleep(3600)`

Comment (strega-nil) [reply scripts/cmake/vcpkg_acquire_msys.cmake:95]: I mean you can do:

```cmake
_execute_process(
    COMMAND ${PATH_TO_ROOT}/usr/bin/bash.exe --noprofile --norc -c "PATH=/usr/bin;pacman -Syuu --noconfirm --disable-download-timeout --overwrite '*'"
    TIMEOUT 3600
)
```

Comment (Voskrese) [reply scripts/cmake/vcpkg_acquire_msys.cmake:95]: https://github.com/appveyor/build-images/blob/350e44f7a27e95859ac8e55bbf0e7f872153c7fa/scripts/Windows/install_msys64.ps1#L33

tested, works locally, not on the VMSS \\cannot remove /usr/bin/pacman.exe (Permission denied)\\
errors associated with junction/symlink and access rights
close

**Pull request closed without merging**